@here/harp-utils/lib/UrlUtils.ts

```diff
@@ -130,7 +130,10 @@ export function getUrlHostAndProtocol(
 
     const match = url.match(urlOriginRe);
     if (!match) {
-        throw new Error(`getUrlHostAndProtocol: unable to parse URL '${url}'`);
+        return {
```

Comment:
Instead of returning empty fields, I think best to try `window.atob(url)` and then re-run the regex and return the protocol / host as on the lines 139